### PR TITLE
PHC and deep misalignment detection (laFedeTesis vol. I)

### PR DESCRIPTION
**Pull request by fede-pe**
This PR includes

- New scipion-em-xmipp protocol for detecting high contrast regions in tomographic reconstruction
- New scipion-em-xmipp protocol misalignment detection in tomographic reconstructions from high-contrast regions
- Test for both procotols
- Changes in subtomogram extraction to allow for downsampling of features
- Needed changes in init, protocols.conf, and utils
- Typos and small fixes

This PR depends on:

https://github.com/I2PC/xmipp/pull/761
https://github.com/I2PC/scipion-em-xmipp/pull/694
https://github.com/I2PC/xmipp/pull/762

### CODE REVIEW

File: xmipptomo/protocols/protocol_deep_misalignment_detection.py [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+
+    # --------------------------- DEFINE param functions ------------------------
+    def _defineParams(self, form):
+        form.addSection(label='Input')
+
+        form.addParam('inputSetOfCoordinates',
+                      PointerParam,
+                      pointerClass=SetOfCoordinates3D,
+                      label='Fiducial 3D coordinates',
+                      help='3D coordinates in dicating the location of the fiducials (gold beads) in the tomogram. '
```
Comment (Vilax) [inline xmipptomo/protocols/protocol_deep_misalignment_detection.py:71]: indicating

**Review by Vilax — APPROVED** [review]

File: xmipptomo/protocols/protocol_deep_misalignment_detection.py [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+from xmipptomo import utils
+
+COORDINATES_FILE_NAME = 'subtomo_coords.xmd'
+BOX_SIZE = 32
+TARGET_SAMPLING_RATE = 6.25
+
+
+class XmippProtDeepDetectMisalignment(EMProtocol, ProtTomoBase):
+    """
+    Wrapper protocol to Xmipp image peak high contrast applied to any volume
```
Comment (albertmena) [inline xmipptomo/protocols/protocol_deep_misalignment_detection.py:46]: Update the help according to the protocol

File: xmipptomo/protocols/protocol_deep_misalignment_detection.py [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+                      BooleanParam,
+                      default=True,
+                      label='Use misalignment threshold?',
+                      help='Threshold to settle if a tomogram presents weak or strong misalignment. If this value is '
+                           'not provided two output set of tomograms are generated, those discarded which present '
+                           'strong misalignment and those which do not. If this value is provided the second group of '
+                           'tomograms is splitted into two, using this threshold to settle if the tomograms present'
+                           'or not a weak misalignment.')
+
+        form.addParam('misaliThr',
```
Comment (albertmena) [inline xmipptomo/protocols/protocol_deep_misalignment_detection.py:103]: Units?

File: xmipptomo/protocols/protocol_deep_misalignment_detection.py [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+
+class XmippProtDeepDetectMisalignment(EMProtocol, ProtTomoBase, XmippProtocol):
+    """
+    Wrapper protocol to Xmipp image peak high contrast applied to any volume
+    """
+
+    _label = 'detect misalignment from fiducials'
+    _devStatus = BETA
+    _conda_env = 'xmipp_DLTK_v1.0'
+
```
Comment (albertmena) [inline xmipptomo/protocols/protocol_deep_misalignment_detection.py:52]: Please add a _possibleOutputs with the possible objects

Comment (albertmena) [reply xmipptomo/protocols/protocol_deep_misalignment_detection.py:52]: Also for the other protocols affected by thisPR

File: xmipptomo/protocols/protocol_deep_misalignment_detection.py [hunk 5]
```diff
@@ -0,0 +1,426 @@
...
+                break
+
+            subtomoPathList.append(subtomoPath)
+            counter += 1
+
+        return subtomoPathList
+
+    # --------------------------- INFO functions ----------------------------
+    def _summary(self):
+        summary = []
```
Comment (albertmena) [inline xmipptomo/protocols/protocol_deep_misalignment_detection.py:419]: Could you provide any stadistic to know about the fiducial alignment?

**Review by albertmena — COMMENTED** [review]
Great job! just minor changes proposed

File: xmipptomo/protocols/protocol_deep_misalignment_detection.py (hunk 3 above)
Comment (fede-pe) [reply xmipptomo/protocols/protocol_deep_misalignment_detection.py:103]: It is an score parameter, i added the range value

File: xmipptomo/protocols/protocol_deep_misalignment_detection.py (hunk 5 above)
Comment (fede-pe) [reply xmipptomo/protocols/protocol_deep_misalignment_detection.py]: done

File: xmipptomo/protocols/protocol_deep_misalignment_detection.py (hunk 4 above)
Comment (fede-pe) [reply xmipptomo/protocols/protocol_deep_misalignment_detection.py:57]: done

File: xmipptomo/protocols/protocol_deep_misalignment_detection.py (hunk 2 above)
Comment (fede-pe) [reply xmipptomo/protocols/protocol_deep_misalignment_detection.py]: done

**Review by albertmena — APPROVED** [review]

**Pull request merged**